src/mp_game_settings.cpp

```diff
@@ -171,23 +184,35 @@ void mp_game_settings::update_addon_requirements(const config & cfg) {
 
 	mp_game_settings::addon_version_info new_data(cfg);
 
+	std::map<std::string, addon_version_info>::iterator it = addons.find(cfg["id"].str());
 	// Check if this add-on already has an entry as a dependency for this scenario. If so, try to reconcile their version info,
 	// by taking the larger of the min versions. The version should be the same for all WML from the same add-on...
-	std::map<std::string, addon_version_info>::iterator it = addons.find(cfg["id"].str());
 	if (it != addons.end()) {
 		addon_version_info & addon = it->second;
 
-		if (new_data.version) {
-			if (!addon.version || (*addon.version != *new_data.version)) {
+		// an add-on can contain multiple types of content
+		// for example, an era and a scenario
+		for(const auto& content : new_data.content.child_range("content")) {
+			config& c = addon.content.add_child("content");
+			c["id"] = content["id"].str();
+			c["type"] = content["type"].str();
+		}
+
+		if(new_data.required) {
+			addon.required = true;
+
+			if (new_data.version) {
+				if (!addon.version || (*addon.version != *new_data.version)) {
+					WRN_NG << "Addon version data mismatch -- not all local WML has same version of '" << cfg["id"].str() << "' addon.\n";
+				}
+			}
+			if (addon.version && !new_data.version) {
 				WRN_NG << "Addon version data mismatch -- not all local WML has same version of '" << cfg["id"].str() << "' addon.\n";
 			}
-		}
-		if (addon.version && !new_data.version) {
-			WRN_NG << "Addon version data mismatch -- not all local WML has same version of '" << cfg["id"].str() << "' addon.\n";
-		}
-		if (new_data.min_version) {
-			if (!addon.min_version || (*new_data.min_version > *addon.min_version)) {
-				addon.min_version = *new_data.min_version;
+			if (new_data.min_version) {
+				if (!addon.min_version || (*new_data.min_version > *addon.min_version)) {
+					addon.min_version = *new_data.min_version;
```

Comment:
not having a min_version specifies currently means, "require the latest version", so not specifying min_version should take precedence over any specified min_version. (for addons with require =yes.)

Reply:
Wouldn't this be incorrect currently as well, then?  The add-on version checking logic being done wasn't changed.

Reply:
> Wouldn't this be incorrect currently as well, then?

Hmm yes

Reply:
Is it alright if I open an issue and fix this in a separate PR, in that case?

Reply:
i guess

Reply:
Created  #5004.